internal/pkg/api/handleEnroll.go

```diff
@@ -242,7 +282,7 @@ func invalidateAPIKey(ctx context.Context, zlog zerolog.Logger, bulker bulk.Bulk
 LOOP:
 	for {
 
-		_, err := bulker.APIKeyRead(ctx, apikeyID, false)
+		_, err := bulker.APIKeyRead(ctx, apikeyID, true)
```

Comment:
I found that the test works if I use the `withOwner` flag in `APIKeyRead`
`// WithOwner - flag to query api keys owned by the currently authenticated user.`

```
{"log.level":"debug","ecs.version":"1.6.0","service.name":"fleet-server","http.request.id":"5492a01d-efc9-436f-b77f-3c45245236eb","server.address":"","mod":"enroll","fleet.enroll.apikey.id":"tEvjs4gBdXQ3jhGbSbnv","EnrollmentId":"123456","AgentId":"a3a9c74d-3a16-4963-94fe-858fd7c81c3c","APIKeyID":"t0vjs4gBdXQ3jhGbU7kM","@timestamp":"2023-06-13T08:33:20.386Z","message":"Invalidate old api key and remove existing agent with the same enrollment_id"}
{"log.level":"debug","ecs.version":"1.6.0","service.name":"fleet-server","http.request.id":"5492a01d-efc9-436f-b77f-3c45245236eb","server.address":"","mod":"enroll","fleet.enroll.apikey.id":"tEvjs4gBdXQ3jhGbSbnv","fleet.apikey.id":"t0vjs4gBdXQ3jhGbU7kM","error.message":"[404 Not Found] {\"api_keys\":[]}: api key not found","@timestamp":"2023-06-13T08:33:20.388Z","message":"api key read"}

{"log.level":"debug","ecs.version":"1.6.0","service.name":"fleet-server","http.request.id":"5492a01d-efc9-436f-b77f-3c45245236eb","server.address":"","mod":"enroll","fleet.enroll.apikey.id":"tEvjs4gBdXQ3jhGbSbnv","fleet.apikey.id":"t0vjs4gBdXQ3jhGbU7kM","@timestamp":"2023-06-13T08:33:21.409Z","message":"api key read"}
{"log.level":"debug","ecs.version":"1.6.0","service.name":"fleet-server","http.request.id":"5492a01d-efc9-436f-b77f-3c45245236eb","server.address":"","mod":"enroll","fleet.enroll.apikey.id":"tEvjs4gBdXQ3jhGbSbnv","fleet.apikey.id":"t0vjs4gBdXQ3jhGbU7kM","@timestamp":"2023-06-13T08:33:21.409Z","message":"api key invalidate"}
{"error_count":0,"invalidated_api_keys":["t0vjs4gBdXQ3jhGbU7kM"],"previously_invalidated_api_keys":[]}
{"log.level":"info","ecs.version":"1.6.0","service.name":"fleet-server","http.request.id":"5492a01d-efc9-436f-b77f-3c45245236eb","server.address":"","mod":"enroll","fleet.enroll.apikey.id":"tEvjs4gBdXQ3jhGbSbnv","fleet.apikey.id":"t0vjs4gBdXQ3jhGbU7kM","dur":1304.635082,"@timestamp":"2023-06-13T08:33:21.691Z","message":"invalidated apiKey"}
{"log.level":"info","ecs.version":"1.6.0","service.name":"fleet-server","http.request.id":"5492a01d-efc9-436f-b77f-3c45245236eb","server.address":"","mod":"enroll","fleet.enroll.apikey.id":"tEvjs4gBdXQ3jhGbSbnv","fleet.agent.id":"a3a9c74d-3a16-4963-94fe-858fd7c81c3c","@timestamp":"2023-06-13T08:33:21.987Z","message":"agent record deleted"}
{"log.level":"info","ecs.version":"1.6.0","service.name":"fleet-server","http.request.id":"5492a01d-efc9-436f-b77f-3c45245236eb","server.address":"","mod":"enroll","fleet.enroll.apikey.id":"tEvjs4gBdXQ3jhGbSbnv","fleet.agent.id":"66d97083-db90-4ab9-b16a-6b6e12676d46","fleet.policy.id":"eaaa69f5-e3bc-4e4f-bbb0-6cd10717d19e","fleet.access.apikey.id":"ukvjs4gBdXQ3jhGbXLlE","http.response.body.bytes":410,"event.duration":2294719000,"@timestamp":"2023-06-13T08:33:22.35Z","message":"Elastic Agent successfully enrolled"}
    fleet_integration_test.go:614: old agent not found as expected
    fleet_integration_test.go:606: could not clean up first agent
--- PASS: Test_Agent_Enrollment_Id (6.32s)
```

Reply:
Sanity check: What is the "user" in this context? Is it the agent? Or is it the Fleet Server? If it's Fleet Server itself does this have any implications when there is more than one Fleet Server?

Reply:
It is Fleet Server, according to the error message: `action [cluster:admin/xpack/security/api_key/get] is unauthorized for service account [elastic/fleet-server],`
I think it shouldn't be an issue with more than one FS as they are using the same service account.